packages/py-moose-lib/moose_lib/dmv2/materialized_view.py

```
@@ -142,3 +249,158 @@ def __init__(
+        # Capture source file from stack trace
+        if not isinstance(self.metadata, dict):
+            self.metadata = {}
+        if "source" not in self.metadata:
+            source_file = get_source_file_from_stack()
+            if source_file:
+                self.metadata["source"] = {"file": source_file}
+
+        if self.name in _materialized_views:
+            raise ValueError(
+                f"RefreshableMaterializedView with name {self.name} already exists"
+            )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)